msbuild/Xamarin.iOS.Tasks.Windows/Xamarin.iOS.Windows.After.targets

```diff
@@ -27,5 +31,43 @@ Copyright (C) 2011-2013 Xamarin. All rights reserved.
 		<RemoveDir SessionId="$(BuildSessionId)" Condition="'$(MtouchTargetsEnabled)' == 'true'" Directories="$(OutputPath);$(IntermediateOutputPath)" RemoveAppDir="$(RemoveAppDir)" ContinueOnError="true" />
 	</Target>
 
+	<!-- Target meant to run locally (Windows), as part of a remote build, to calculate the file hashes of the OutputPath for .dll and .pdb files, generating a resulting file with the list -->
+	<!-- We don't want the remote build to fail in case this target fails, so we treat erros as warnings -->
+	<Target Name="CalculateLocalOuputHashes" Condition="'$(IsRemoteBuild)' == 'true' And '$(IsMacEnabled)' == 'true'" BeforeTargets="BeforeDisconnect" Inputs="$(OutputPath)" Outputs="$(IntermediateOutputPath)$(LocalOutputHashesFileName)">
+		<ItemGroup>
+			<FileExtension Include=".dll"/>
+			<FileExtension Include=".pdb"/>
+		</ItemGroup>
+
+		<!-- This task will run locally since we don't pass the SessionId -->
+		<CalculateFileHashes WorkingDirectory="$(OutputPath)" FileExtensions="@(FileExtension)" TargetHashesFile="$(IntermediateOutputPath)$(LocalOutputHashesFileName)" ContinueOnError="WarnAndContinue"/>
+	</Target>
+
+	<!-- Target meant to run part locally (Windows) and part remotely (Mac), as part of a remote build, to analyze which files from the remote OutputPath have changed, compared to the local OutputPath, and to copy those files back to the OutputPath in Windows -->
+	<!-- We don't want the remote build to fail in case this target fails, so we treat erros as warnings -->
+	<Target Name="CopyChangedOutputFilesFromMac" Condition="'$(IsRemoteBuild)' == 'true' And '$(IsMacEnabled)' == 'true'" AfterTargets="CalculateLocalOuputHashes">
```

Comment:
I think you need to exclude library projects here.

Example binlog where tests failed because an unexpected warning was shown: 

https://vsdrop.corp.microsoft.com/file/v1/macios/device-tests/20250424.10/11465088/windows_integrationwindows-1/;tests/dotnet/BundleStructure/iOS/log-build-20250423_211021.binlog

(search for "Unable to analyze file changes" and the warnings show up)

Reply:
Done